src/software/gui/full_system/widgets/full_system_gui.cpp

```diff
@@ -96,3 +107,29 @@ void FullSystemGUI::updateDrawViewArea()
         main_widget->ai_visualization_graphics_view->setViewArea(view_area.value());
     }
 }
+
+void FullSystemGUI::updateDataPerSecondLCD()
+{
+    if (auto data_received_per_second =
+            data_received_per_second_buffer->popLeastRecentlyAddedValue())
+    {
+        int data_received_int = (int)data_received_per_second.value();
+        // Formatting code to display fixed format Ex. 60.00
+        main_widget->data_received_lcd->display(
+            QString("%1").arg(data_received_int, 2, 10, QChar(' ')) + "." +
+            QString("%1").arg(
+                qRound((data_received_per_second.value() - data_received_int) * 100), 2,
+                10, QChar('0')));
+    }
```

Comment:
We probably don't need the extra decimal points of precision on the rates, we can just show the "integer" version and it would be just as informative